micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -0,0 +1,94 @@
+#![allow(dead_code)]
+use std::cell::RefCell;
+
+use esp_idf_svc::{
+    nvs::{EspCustomNvs, EspCustomNvsPartition, EspNvs},
+    sys::EspError,
+};
+use thiserror::Error;
+
+use crate::common::provisioning::storage::{
+    RobotCredentialStorage, RobotCredentials, WifiCredentialStorage, WifiCredentials,
+};
+
+#[derive(Error, Debug)]
+pub enum NVSStorageError {
+    #[error(transparent)]
+    EspError(#[from] EspError),
+    #[error("nvs key {0} is absent")]
+    NVSKeyAbsent(&'static str),
+}
+
+pub struct NVSStorage {
```

Comment:
I guess if the other is `MemoryCredentialsStorage` then this one should be `NVSCredentialsStorage`, despite my misgivings about the `MemoryCredentialsStorage` name. If another scheme for `MemoryCredentialsStorage` is found, then this should be renamed analogously.

Reply:
see comment about renaming it to RAMStorage